slight/src/commands/add.rs

```diff
@@ -0,0 +1,84 @@
+use std::{
+    fs::{create_dir_all, remove_dir_all, File},
+    io::{self, ErrorKind},
+};
+
+use anyhow::Result;
+
+const GITHUB_URL: &str = "https://github.com/deislabs/spiderlightning/releases/download";
+
+const KV_DOWNLOADS: [&str; 3] = ["kv", "types", "resources"];
+const CONFIGS_DOWNLOADS: [&str; 2] = ["configs", "types"];
+const EVENTS_DOWNLOADS: [&str; 4] = ["events", "event-handler", "types", "resources"];
+const HTTP_DOWNLOADS: [&str; 4] = ["http", "http-handler", "http-types", "types"];
+const LOCKD_DOWNLOADS: [&str; 3] = ["lockd", "types", "resources"];
+const MQ_DOWNLOADS: [&str; 3] = ["mq", "types", "resources"];
+const PUBSUB_DOWNLOADS: [&str; 3] = ["pubsub", "types", "resources"];
+
+pub async fn handle_add(what_to_add: &str) -> Result<()> {
+    let (interface, release, folder_name) = if !what_to_add.contains('@') {
+        panic!("invalid use: to download an interface, say `slight add <interface-name>@<release-tag>`");
+        // TODO: In the future, let's support omitting the release tag to download the latest release
+    } else {
+        let find_at = what_to_add.find('@').unwrap();
+        // ^^^ fine to unwrap, we are guaranteed to have a '@' at this point.
+        (
+            &what_to_add[..find_at],
+            &what_to_add[find_at + 1..],
+            what_to_add.replace('@', "_"),
+        )
+    };
+
+    match what_to_add {
+        _ if interface.eq("kv")
```

Comment:
I wonder if there is a way to avoid hard cording all the interface names here, and below. I don't know the answer myself :( Will need to think about it. Otherwise everything lgtm!

Reply:
Yeah~ It's annoying 'cause every time that we add a new interface or a new dependency to an interface, we'll have to modify the `add` command...

In any case, considering this is just a "hack" of a more solidified `add` command for when the wasm registry is more fleshed out, I figured it would be alright.